### PR TITLE
terraform-vpc-peerings: extended early exit

### PR DESCRIPTION
**Pull request by chassing**
Add extended early exit to terraform-vpc-peerings

Review commit by commit:
- **:recycle: add type hints**
- **:sparkles: add extended early exit**

### CODE REVIEW

File: reconcile/terraform_vpc_peerings.py [hunk 1]
```diff
@@ -60,11 +67,11 @@ def _get_default_management_account(
...
-    account: dict[str, Any],
-    cluster: dict[str, Any],
-    ocm: Optional[OCM],
-    provided_assume_role: Optional[str],
-) -> Optional[dict[str, Any]]:
+    account: Mapping[str, Any],
+    cluster: Mapping[str, Any],
+    ocm: OCM | None,
+    provided_assume_role: str | None,
+) -> Mapping[str, Any] | None:
```
Comment (hemslo) [inline reconcile/terraform_vpc_peerings.py:74]: return type can be concrete type `dict[str, Any] | None`, generic in, concrete out.

File: reconcile/terraform_vpc_peerings.py [hunk 2]
```diff
@@ -97,12 +104,12 @@ def _build_infrastructure_assume_role(
...
-    accepter_connection: dict[str, Any],
-    accepter_cluster: dict[str, Any],
-    ocm: Optional[OCM],
-) -> tuple[str, dict[str, Any], dict[str, Any]]:
+    requester_connection: Mapping[str, Any],
+    requester_cluster: Mapping[str, Any],
+    accepter_connection: Mapping[str, Any],
+    accepter_cluster: Mapping[str, Any],
+    ocm: OCM | None,
+) -> tuple[str, Mapping[str, Any], Mapping[str, Any]]:
```
Comment (hemslo) [inline reconcile/terraform_vpc_peerings.py:112]: similar here `tuple[str, dict[str, Any], dict[str, Any]]`

File: reconcile/terraform_vpc_peerings.py [hunk 3]
```diff
@@ -655,30 +684,71 @@ def run(
...
+
+
+def runner(
+    dry_run: bool,
+    tf: terraform.TerraformClient,
+    enable_deletion: bool = False,
+    defer: Callable | None = None,
+) -> ExtendedEarlyExitRunnerResult:
+    if defer:
+        defer(tf.cleanup)
```
Comment (hemslo) [inline reconcile/terraform_vpc_peerings.py:733]: `defer` should be called right after `tf` https://github.com/app-sre/qontract-reconcile/pull/4381/files#diff-79f381a7c684c7a6788d3410826abaa6412cd5767ee6694e8e06154fe8e0268fR674-R682, then it's not needed to be part of `RunnerParams`

File: reconcile/terraform_vpc_peerings.py (hunk 1 above)
Comment (chassing) [reply reconcile/terraform_vpc_peerings.py]: Fixed in [daa3332](https://github.com/app-sre/qontract-reconcile/pull/4381/commits/daa3332b0ef1a46304faafb082ce85dd382be201)

File: reconcile/terraform_vpc_peerings.py (hunk 3 above)
Comment (chassing) [reply reconcile/terraform_vpc_peerings.py]: Fixed in [daa3332](https://github.com/app-sre/qontract-reconcile/pull/4381/commits/daa3332b0ef1a46304faafb082ce85dd382be201)

**Review by hemslo — APPROVED** [review]
LGTM! Another faster integration!

**Pull request merged**